state/pruner.go

```diff
@@ -164,23 +153,16 @@ func (p *Pruner) SetCompanionRetainHeight(height int64) error {
 	if !p.checkHeightBound(height) {
 		return ErrInvalidHeightValue
 	}
-	currentCompanionRetainHeight, err := p.stateStore.GetCompanionBlockRetainHeight()
-	if err != nil {
-		if !errors.Is(err, ErrKeyNotFound) {
-			return err
-		}
-		currentCompanionRetainHeight = height
+	currentMinHeight := p.findMinRetainHeight()
+
+	if height < currentMinHeight {
+		return ErrPrunerCannotLowerRetainHeight
 	}
-	currentAppRetainHeight, err := p.stateStore.GetApplicationRetainHeight()
-	noAppRetainHeight := false
+	_, err := p.stateStore.GetCompanionBlockRetainHeight()
```

Comment:
What is this get operation for? Checking if the key already exists in the database?

Reply:
no, it seems mostly to check if there's no error with the key in the database (e.g. < 0). Do we really need this ?

Reply:
You are totally right, this was left over from the version where I checked the new height of the dc companion is not lower than its own previous height, which I deemed not needed anymore because of the comparison against the minimum height ..allowing it to set a lower height in case it's own height was not the miminum.